app/Keycloak/Service/CreateClientFlow.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Keycloak\Service;
+
+use App\Domain\Integrations\Integration;
+use App\Keycloak\ClientCollection;
+use App\Keycloak\Config;
+use App\Keycloak\Exception\KeyCloakApiFailed;
+use App\Keycloak\ScopeConfig;
+use Psr\Log\LoggerInterface;
+
+final readonly class CreateClientFlow
+{
+    public function __construct(
+        private ApiClient $client,
+        private Config $config,
+        private ScopeConfig $scopeConfig,
+        private LoggerInterface $logger
+    ) {
+    }
+
+    public function createClientsForIntegration(Integration $integration): ClientCollection
```

Comment:
Can this become part of the `ApiClient`? I don't think there is added value of having this extra class.

Reply:
I would prefer not too because later on I plan to also add the logic to save the Eloquent model to the database for each client. 
I personally like to have several more smaller classes than 1 large class, but if you want me too I will merge them.

Reply:
A dedicated class per action (create, update, activate, block) is fine for me, but I would search for a better name than `CreateClientFlow::createClientsForIntegration`. Maybe you can apply some naming conventions like a command handler `CreateClientHandler:handle`

Reply:
great idea, will change the naming!